kie-server-parent/kie-server-services/kie-server-services-kafka/src/main/java/org/kie/server/services/jbpm/kafka/KafkaServerExtension.java

```diff
@@ -260,4 +244,26 @@ public void disposeContainer(String id, KieContainerInstance kieContainerInstanc
     public <T> T getAppComponents(Class<T> serviceType) {
         return null;
     }
+    
+    protected final Map<String, Object> getConsumerProperties() {
+        return Collections.unmodifiableMap(consumerProperties);
+    }
+
+    protected final Map<String, Object> getProducerProperties() {
+        return Collections.unmodifiableMap(producerProperties);
+    }
+
+    protected final void initProperties() {
+        for (Map.Entry<Object, Object> entry : System.getProperties().entrySet()) {
+            String key = entry.getKey().toString();
+            if (key.startsWith(KAFKA_EXTENSION_PREFIX)) {
+                String propName = key.substring(KAFKA_EXTENSION_PREFIX.length());
+                if (ConsumerConfig.configNames().contains(propName)) {
+                    consumerProperties.put(propName, entry.getValue());
+                } else if (ProducerConfig.configNames().contains(propName)) {
+                    producerProperties.put(propName, entry.getValue());
```

Comment:
I think these "ifs" shouldn't be nested, as some properties (`CommonClientConfigs`) belong to both consumer and producer.

Reply:
good point, changed